internal/controller/promotions/promotions.go

```diff
@@ -67,7 +67,7 @@ type reconciler struct {
 		types.NamespacedName,
 	) (*kargoapi.Stage, error)
 
-	promoteFn func(context.Context, kargoapi.Promotion) (*kargoapi.PromotionStatus, error)
+	promoteFn func(context.Context, kargoapi.Promotion, *kargoapi.Freight) (*kargoapi.PromotionStatus, error)
```

Comment:
As @krancour pointed out, I think it makes more sense to show the status (alias in here) of freight at the time the user queries events. (Because there can be a possibilities to mislead wrong reference). So I think it would be better to remove Freight alias from event itself.

Reply:
My mention that alias was mutable was an fyi... if you think there's value to recording the alias by which the freight was known at the time of the promotion, I defer you your judgement. As long as the actual freight name is recorded as well, all is good.